### PR TITLE
Integration test for clean cmd

### PR DESCRIPTION
**Pull request by armaniferrante**

### CODE REVIEW

File: src/promise-fs.js [hunk 1]
```diff
@@ -148,6 +148,25 @@ async function writeArtifact(artifact) {
   await writeFile(p, JSON.stringify(artifact, null, 2));
 }
 
+/**
+ * @returns the list of directories containing a file named, filename.
+ */
+async function findDirectories(dir, filename) {
```
Comment (armaniferrante) [inline src/promise-fs.js]: just moved this out into its own function

File: src/promise-fs.js [hunk 2]
```diff
@@ -148,6 +148,25 @@ async function writeArtifact(artifact) {
   await writeFile(p, JSON.stringify(artifact, null, 2));
 }
 
+/**
+ * @returns the list of directories containing a file named, filename.
```
Comment (willscott) [inline src/promise-fs.js]:
```suggestion
 * @returns the list of directories containing a file named `filename`.
```

File: src/promise-fs.js [hunk 3]
```diff
@@ -148,6 +148,25 @@ async function writeArtifact(artifact) {
...
+async function findDirectories(dir, filename) {
+  let _findDirectories = async function(dir) {
+    let files = await readDir(dir);
+    let found = [];
+    await Promise.all(files.map(async (f) => {
+      if (fs.statSync(path.join(dir, f)).isDirectory()) {
+        const newCrates = await _findDirectories(path.join(dir, f));
+        found = found.concat(newCrates);
+      } else if (f === filename) {
+        found.push(dir);
```
Comment (willscott) [inline src/promise-fs.js]: push dir, or push `path.join(dir, filename)`?

Comment (armaniferrante) [reply src/promise-fs.js]: We want to use `dir`. 

When we use this to search for the path to rust crates, filename in is `Cargo.toml`. However, what we want is the directory containing the toml file.

**Pull request merged**